src/traces/heatmap/plot.js

```diff
@@ -18,6 +18,50 @@ var LINE_SPACING = alignmentConstants.LINE_SPACING;
 
 var labelClass = 'heatmap-label';
 
+// Pixelated image rendering
+// The actual declaration is prepended with fallbacks for older browsers.
+// https://developer.mozilla.org/en-US/docs/Web/CSS/image-rendering
+// https://caniuse.com/?search=image-rendering
+var pixelatedImageCSS = [
+    'image-rendering: optimizeSpeed',
+    'image-rendering: -moz-crisp-edges',
+    'image-rendering: -o-crisp-edges',
+    'image-rendering: -webkit-optimize-contrast',
+    'image-rendering: optimize-contrast',
+    'image-rendering: crisp-edges',
+    'image-rendering: pixelated'
+];
```

Comment:
Could we possibly reuse these from `src/traces/image/constants.js` or vice versa?

Reply:
Yes I thought about this, but then wondered what if one creates a custom bundle with heatmap trace but without image trace (or vice versa) ? Could it break things ? If yes, we could put it in `src/constants/pixelated_image.js` instead ?

We could also put the detection function in `lib` so that it can be reused as well.  What do you think ?

Reply:
That shouldn't break the custom bundles. But yes, `src/lib` could be a good central place to have it.

Reply:
Ok so I will put both the CSS array and the detection function in `src/lib` tomorrow. And will make `src/traces/image/plot.js` re-use it.

Reply:
Thank you!

Reply:
So I put them both into `src/lib/index.js`. Not sure it was the best move as I see there is a circular dependency now :/, I'm gonna fix it. 

Also, if it's preferable to have the function in a specific file ie. `src/lib/supports_pixelated_image.js`, don't hesitate! Same for the CSS declarations array and the style string, ie. could be in `src/constants/pixelated_image.js`.

Reply:
So at first I put them both into `src/lib/index.js` but I created a circular dependency so I decided to put them in dedicated files. Don't hesitate to tell me if there is another (preferred) way to do that !